packages/vscode-extension/src/panels/WorkspaceConfigController.ts

```diff
@@ -1,86 +1,73 @@
-import { EventEmitter } from "stream";
 import { ConfigurationChangeEvent, workspace, Disposable } from "vscode";
-import {
-  PanelLocation,
-  WorkspaceConfig,
-  WorkspaceConfigProps,
-  WorkspaceConfigEventMap,
-  WorkspaceConfigEventListener,
-} from "../common/WorkspaceConfig";
 import { getTelemetryReporter } from "../utilities/telemetry";
+import { PanelLocation, WorkspaceConfiguration } from "../common/State";
+import { StateManager } from "../project/StateManager";
+import { disposeAll } from "../utilities/disposables";
 
-export class WorkspaceConfigController implements Disposable, WorkspaceConfig {
-  private config: WorkspaceConfigProps;
-  private eventEmitter = new EventEmitter();
-  private configListener: Disposable | undefined;
+export class WorkspaceConfigController implements Disposable {
+  private disposables: Disposable[] = [];
 
-  constructor() {
+  constructor(private stateManager: StateManager<WorkspaceConfiguration>) {
     const configuration = workspace.getConfiguration("RadonIDE");
-    this.config = {
+    const workspaceConfig = {
       panelLocation: configuration.get<PanelLocation>("panelLocation")!,
       showDeviceFrame: configuration.get<boolean>("showDeviceFrame")!,
       stopPreviousDevices: configuration.get<boolean>("stopPreviousDevices")!,
     };
 
-    this.configListener = workspace.onDidChangeConfiguration((event: ConfigurationChangeEvent) => {
-      if (!event.affectsConfiguration("RadonIDE")) {
-        return;
-      }
-      const config = workspace.getConfiguration("RadonIDE");
+    this.stateManager.setState(workspaceConfig);
 
-      const newConfig = {
-        panelLocation: config.get<PanelLocation>("panelLocation")!,
-        showDeviceFrame: config.get<boolean>("showDeviceFrame")!,
-        stopPreviousDevices: config.get<boolean>("stopPreviousDevices")!,
-      };
+    this.disposables.push(
+      workspace.onDidChangeConfiguration((event: ConfigurationChangeEvent) => {
```

Comment:
```suggestion
      workspace.onDidChangeConfiguration(this.onConfigurationChange)
```
while we're here, we could clean this up a bit.